src/parser.rs

```diff
@@ -884,6 +946,22 @@ impl Get for Attributes {
     }
 }
 
+fn read_text(parser: &mut Reader) -> Result<Option<String>, Error> {
+    match try!(parser.next()) {
+        Characters(mut text) => {
+            Ok(if let Some(ret) = try!(read_text(parser)) {
+                text.push_str(&ret);
+                Some(text)
+            } else {
+                Some(text)
+            })
+        }
+        EndElement { .. } => Ok(None),
+        StartElement { .. } => Err(mk_error!("Unexpected StartElement", parser)),
```

Comment:
This would preferably be consistent with other "unexpected element" errors (e.g. featuring the element's name).